crypto/x509/x509_vfy.c

```diff
@@ -986,15 +1011,142 @@ static int check_revocation(X509_STORE_CTX *ctx)
     }
     for (i = 0; i <= last; i++) {
         ctx->error_depth = i;
-        ok = check_cert(ctx);
-        if (!ok)
-            return ok;
+#ifndef OPENSSL_NO_OCSP
+        if (ctx->param->flags & X509_V_FLAG_OCSP_RESP_CHECK) {
+            ok = check_cert_ocsp_resp(ctx);
+
+            /*
+             * If the client receives a "ocsp_response_list" that does not
+             * contain a response for one or more of the certificates in the
+             * completed certificate chain, the client SHOULD attempt to
+             * validate the certificate using an alternative retrieval method,
+             * such as downloading the relevant CRL;
+             * If the OCSP response received from the server does not result
+             * in a definite "good" or "revoked" status, it is inconclusive.
+             * A TLS client in such a case MAY check the validity of the server
+             * certificate through other means, e.g., by directly querying the
+             * certificate issuer.
+             */
+            if (ok == V_OCSP_CERTSTATUS_REVOKED) {
+                return verify_cb_ocsp(ctx, X509_V_ERR_CERT_REVOKED);
+            }
+
+            if (ok == V_OCSP_CERTSTATUS_GOOD)
+                continue;
+
+            /*
+             * Check the validity of the server certificate through other means,
+             * e.g., by directly querying the certificate issuer.
```

Comment:
For clarity, please prepend the sentence by, e.g.: 
```
Unless CRL-based cert status checking is enabled,
which should be tried first before doing an OCSP query,
```

Reply:
Done

Reply:
It does not seem to be pushed/committed?

Reply:
Is committed and pushed